### PR TITLE
Improve the 'Suppress JIT Optimizations' debugger help topics

### PR DESCRIPTION
**Pull request by gregg-miskelly**
This makes several improvements to the 'Suppress JIT Optimizations' help topics:
1. The help topic was written back when the default state of the option was checked, but that hasn't been true for several releases now, and the help topic was never updated.
2. The help topic lacked a bunch of details that are important to understand what the option really does. So I tried to explain everything.
3. There wasn't a link from the Tools->Options->Debugging->General options page.

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
cc @Mikejo5000

File: docs/debugger/jit-optimization-and-debugging.md [hunk 1]
```diff
@@ -25,18 +25,30 @@ ms.workload:
...
+
+In the .NET ecosystem, code is turned from source to CPU instructions in a two-step process: first, the C# compiler converts the text you type to an intermediate binary form called MSIL and writes this out to .dll files. Later, the .NET Runtime converts this MSIL to CPU instructions. Both steps can optimize to some degree, but the second step performed by the .NET Runtime performs the more significant optimizations.
+
+**The 'Suppress JIT optimization on module load' option:** The debugger exposes an option that controls what happens when a DLL that is compiled with optimizations enabled loads inside of the target process. If this option is unchecked (the default state), then when the .NET Runtime compiles the MSIL code into CPU code, it leaves the optimizations enabled. If the option is checked, then the debugger requests that optimizations be disabled.
+
+The **Suppress JIT optimization on module load (Managed only)** option can be found on the **General** page under the **Debugging** node in the **Options** dialog box.
```
Comment (Mikejo5000) [inline docs/debugger/jit-optimization-and-debugging.md]: Gregg, the second reference to this option includes "(Managed only)". It looks like the first reference in the subheading should include this also...

**Comment by Mikejo5000** [comment]
@gregg-miskelly Gregg, looks good, I commited some style and formatting edits directly to the PR. Mostly the edits were switching to present tense when possible. I left one comment for you to check.

File: docs/debugger/jit-optimization-and-debugging.md (hunk 1 above)
Comment (gregg-miskelly) [reply docs/debugger/jit-optimization-and-debugging.md]: Agreed. It would be better with '(Managed only)'

**Pull request merged**